WalletWasabi.Gui/CrashReport/CrashReporter.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Diagnostics;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Gui.CrashReport
+{
+	public class CrashReporter
+	{
+		private const int MaxRecursiveCalls = 5;
+		public int Attempts { get; private set; }
+		public string ExceptionString { get; private set; } = null;
+		public bool IsReport { get; private set; }
+		public bool HadException { get; private set; }
+		public bool IsInvokeRequired => !IsReport && HadException;
+		public SerializableException SerializedException { get; private set; }
+
+		public void InvokeCrashReport()
+		{
+			var exceptionString = ExceptionString;
+			var prevAttempts = Attempts;
+
+			if (prevAttempts >= MaxRecursiveCalls)
+			{
+				Logger.LogCritical($"The crash helper has been called {MaxRecursiveCalls} times. Will not continue to avoid recursion errors.");
+				return;
+			}
+
+			var args = $"crashreport -attempt={prevAttempts + 1} -exception={exceptionString}";
```

Comment:
Can `exceptionString` be null? How come there is no escaping for `-exception={exceptionString}?` Is it guaranteed that it's a string without spaces?

Reply:
Fixed